test/Whitelist.test.ts

```diff
@@ -50,6 +63,28 @@ contract('Whitelist', ([owner, random, newOwner]) => {
     })
   })
 
+  describe('Whitelist otoken', () => {
+    it('should revert whitelisting an otoken from sender other than Otoken Factory', async () => {
+      await expectRevert(whitelist.whitelistOtoken(otoken.address, {from: owner}), 'Sender is not Otoken Factory')
+    })
+
+    it('should whitelist an otoken from Otoken Factory address', async () => {
```

Comment:
Can we add a test to ensure that after Factory Address has changed in the addressBook, the new oToken can still be whitelisted

Reply:
In upgradability, the first deployed address will alway be the caller, as `V1` will delegate call to `V2`, then `V2` will execute, in the execution transaction, `V1` is the `msg.sender`. (V1 is first version of the contract and V2 is an upgrade to a second version).

Reply:
Or no I think I'm wrong in this, will make sure of it, if the contract address change when it make an external call, will add the needed test for that.

Reply:
Sorry my first response is out-of-topic, so when we change the implementation address of the Factory module in the addressBook, nothing change for the whitelist module as the `onlyFactory` modifier always call the addressBook to get the updated implementation address.